cmd/soroban-rpc/lib/preflight/src/ledger_storage.rs

```diff
@@ -186,7 +186,17 @@ impl LedgerStorage {
         }
 
         let entry = LedgerEntry::from_xdr(xdr)?;
-        Ok((entry, expiration_seq))
+        let entry_and_expiration = (entry, expiration_seq);
+
+        if let Ok(expirable_entry) =
+            TryInto::<Box<dyn ExpirableLedgerEntry>>::try_into(&entry_and_expiration)
+        {
+            if expirable_entry.durability() != Persistent {
+                return Err(Error::NotFound);
```

Comment:
I'm not sure I understand this, why don't we do a `has_expired` check here? Seems like every temp entry will be `NotFound` now...